internal/app/machined/pkg/system/services/registryd.go

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package services
+
+import (
+	"context"
+	"io"
+
+	"go.uber.org/zap/zapcore"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system/events"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system/health"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system/runner"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system/runner/goroutine"
+	"github.com/siderolabs/talos/internal/app/machined/pkg/system/services/registry"
+	"github.com/siderolabs/talos/pkg/conditions"
+	"github.com/siderolabs/talos/pkg/logging"
+)
+
+var _ system.HealthcheckedService = (*Registryd)(nil)
+
+// Registryd implements the Service interface. It serves as the concrete type with
+// the required methods.
+type Registryd struct{}
+
+// ID implements the Service interface.
+func (m *Registryd) ID(runtime.Runtime) string {
+	return "registryd"
+}
+
+// PreFunc implements the Service interface.
+func (m *Registryd) PreFunc(context.Context, runtime.Runtime) error {
+	return nil
+}
+
+// PostFunc implements the Service interface.
+func (m *Registryd) PostFunc(runtime.Runtime, events.ServiceState) (err error) {
+	return nil
+}
+
+// Condition implements the Service interface.
+func (m *Registryd) Condition(runtime.Runtime) conditions.Condition {
+	return nil
+}
+
+// DependsOn implements the Service interface.
+func (m *Registryd) DependsOn(runtime.Runtime) []string {
+	return nil
+}
+
+// Runner implements the Service interface.
+func (m *Registryd) Runner(r runtime.Runtime) (runner.Runner, error) {
+	return goroutine.NewRunner(r, "registryd", func(ctx context.Context, r runtime.Runtime, logOutput io.Writer) error {
+		return registry.NewService(
+			"/imagecache",
+			logging.ZapLogger(logging.NewLogDestination(
+				logOutput,
+				zapcore.DebugLevel,
+				logging.WithColoredLevels(),
+			)),
+		).Run(ctx)
+	}, runner.WithLoggingManager(r.Logging())), nil
+}
+
+// HealthFunc implements the HealthcheckedService interface.
+func (m *Registryd) HealthFunc(runtime.Runtime) health.Check {
+	return func(ctx context.Context) error {
+		// TODO: implement me
```

Comment:
we should just do TCP/HTTP check here, should be trivial, e.g. like `kubelet` health check